### PR TITLE
Add gitea

### PR DESCRIPTION
**Pull request by jojomomojo**
- Add gitea as a service
- Specific hostnames on KITT_DOMAIN (formerly matches any domain name)
- don't inherit environment variables, initialize from .env

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -60,11 +55,26 @@ services:
...
+    volumes:
+      - ./etc/gitea:/data
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.gitea.entrypoints=https"
+      - "traefik.http.routers.gitea.rule=Host(`gitea.${KITT_DOMAIN}`)"
+      - "traefik.http.services.gitea.loadbalancer.server.port=3000"
+      - "traefik.tcp.routers.ssh.entrypoints=ssh"
+      - "traefik.tcp.routers.ssh.rule=HostSNI(`*`)"
+      - "traefik.tcp.services.ssh.loadbalancer.server.port=22"
```
Comment (bellaweo) [inline docker-compose.yml:77]: is this port 2222 now?

File: docker-compose.yml [hunk 2]
```diff
@@ -60,11 +55,26 @@ services:
...
+      - "traefik.http.routers.registry.rule=Host(`registry.${KITT_DOMAIN}`)"
       - "traefik.http.services.registry.loadbalancer.server.port=5000"
     volumes:
       - ./etc/registry/data:/var/lib/registry
       - ./etc/registry/auth:/auth
+  gitea:
+    image: gitea/gitea:latest
+    environment:
+      - USER_UID=1000
+      - USER_GID=1000
```
Comment (bellaweo) [inline docker-compose.yml:67]: future feature: detect and generate UID & GID. identify current user running kiki.

**Comment by bellaweo** [comment]
i'm warming up to the idea of having a local gitea available. we should add in foo to run regular backups. ideally in the background. i'm gonna look into it i think.

File: docker-compose.yml (hunk 1 above)
Comment (jojomomojo) [reply docker-compose.yml:77]: The port 22 is on the gitea container side.  Traefik maps its port 2222 to gitea's 22.

File: docker-compose.yml (hunk 2 above)
Comment (jojomomojo) [reply docker-compose.yml:67]: I took this from the examples, not sure if the 1000 ids track what's in the container file-system, or whether it maps to the host user.

**Comment by jojomomojo** [comment]
Let's run postgres as the common database for gitea and drone, then we can back up that one database.

**Comment by jojomomojo** [comment]
> i'm getting this message from registry:
> 
> ```
> No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable.
> ```
> i do not have a htpasswd configured for registry (but the error above may be unrelated to htpasswd?)

I'm not using passwords for local registry yet, so it is open for local dev.    Thinking if we get more fancy, we should look into running sonatype nexus as a general package repo and passthrough cache.  This pushes the theme of local dev, repro everything locally.

**Comment by bellaweo** [comment]
> > i'm getting this message from registry:
> > ```
> > No HTTP secret provided - generated random secret. This may cause problems with uploads if multiple registries are behind a load-balancer. To provide a shared secret, fill in http.secret in the configuration file or set the REGISTRY_HTTP_SECRET environment variable.
> > ```
> > 
> > 
> > i do not have a htpasswd configured for registry (but the error above may be unrelated to htpasswd?)
> 
> I'm not using passwords for local registry yet, so it is open for local dev. Thinking if we get more fancy, we should look into running sonatype nexus as a general package repo and passthrough cache. This pushes the theme of local dev, repro everything locally.

not familiar but sounds interesting!

**Pull request merged**